From: Jari Aalto <jari.aalto@cante.net>
Date: Fri, 5 Mar 2010 11:28:43 +0200
Subject: lwm.c: Use Debian terminal setting

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 lwm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lwm.c b/lwm.c
index f67a663..e351aea 100644
--- a/lwm.c
+++ b/lwm.c
@@ -299,7 +299,7 @@ shell(ScreenInfo * screen, int button, int x, int y) {
 		execl(sh, sh, "-c", command, NULL);
 		fprintf(stderr, "%s: can't exec \"%s -c %s\"\n", argv0, sh,
 			command);
-		execlp("xterm", "xterm", NULL);
+		execlp("x-terminal-emulator", "x-terminal-emulator", NULL);
 		exit(EXIT_FAILURE);
 	case -1:	/* Error. */
 		fprintf(stderr, "%s: couldn't fork\n", argv0);
