From: Jari Aalto <jari.aalto@cante.net>
Date: Tue, 12 Oct 2010 23:46:33 +0300
Subject: disp.c: increase target area for close box

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 disp.c | 17 +++++++++--------
 1 file changed, 9 insertions(+), 8 deletions(-)

diff --git a/disp.c b/disp.c
index 612c4e2..039cc0e 100644
--- a/disp.c
+++ b/disp.c
@@ -140,6 +140,11 @@ expose(XEvent * ev) {
 	}
 }
 
+/* #define in_closing_box(e, quarter)  (e->x > (quarter + 2) && e->x < (3 + 3*quarter) && e->y > quarter && e->y <= 3*quarter) */ /* original */
+/* #define in_closing_box(e, quarter)  (e->x > (quarter + 1) && e->x < (4 + 3*quarter) && e->y > quarter - 1 && e->y <= 3*quarter + 1) */ /* 2 pixels wider in each dir */
+#define in_closing_box(e, quarter)  (e->x > (quarter + 0) && e->x < (5 + 3*quarter) && e->y > quarter - 2 && e->y <= 3*quarter + 2)  /* 4 pixels wider in each dir */
+
+
 static void
 buttonpress(XEvent *ev) {
 	Client *c;
@@ -178,7 +183,7 @@ buttonpress(XEvent *ev) {
 
 		/* The ``box''. */
 		quarter = (border + titleHeight()) / 4;
-		if (e->x > (quarter + 2) && e->x < (3 + 3*quarter) && e->y > quarter && e->y <= 3*quarter) {
+		if (in_closing_box(e, quarter)) {
 			/*Client_Close(c);*/
 			pending = c;
 			mode = wm_closing_window;
@@ -249,10 +254,8 @@ buttonrelease(XEvent *ev) {
 		/* was the button released within the window's box?*/
 		quarter = (border + titleHeight()) / 4;
 		if (pending != NULL &&
-			(e->window == pending->parent) &&
-			(e->x > (quarter + 2) &&
-				e->x < (3 + 3*quarter) &&
-				e->y > quarter && e->y <= 3*quarter))
+		    (e->window == pending->parent) &&
+		    in_closing_box(e, quarter))
 					Client_Close(pending);
 		pending = NULL;
 	} else if (mode == wm_hiding_window) {
@@ -752,9 +755,7 @@ motionnotify(XEvent *ev) {
 			(e->subwindow != c->window) &&
 			mode == wm_idle) {
 			/* mouse moved in a frame we manage - check cursor */
-			if (e->x > (quarter + 2)
-				&& e->x < (3 + 3*quarter)
-				&& e->y > quarter && e->y <= 3*quarter) {
+			if (in_closing_box(e, quarter)) {
 				edge = E_LAST;
 			} else if (e->x <= border && e->y <= border) {
 				edge = ETopLeft;
