From: Kacper Gutowski <klg@mmax.one.pl>
Date: Thu, 14 Apr 2016 12:05:54 +0200
Subject: Minor typo in manual

---
 lwm.man | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lwm.man b/lwm.man
index a527498..23f0600 100644
--- a/lwm.man
+++ b/lwm.man
@@ -70,7 +70,7 @@ exclusively by session managers.
 .SH RESOURCES
 \fILwm\fP understands the following X resources:
 .TP 12
-.B titlefont
+.B titleFont
 font used in window titles
 .TP 12
 .B popupFont
