From fedf6be86fb26a2a426955ef93b5e00704632ba7 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Tue, 9 Mar 2010 11:07:50 +0200
Subject: [PATCH] Imakefile: Link with ICE library


Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 Imakefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Imakefile b/Imakefile
index d09498c..71e5a28 100644
--- a/Imakefile
+++ b/Imakefile
@@ -17,7 +17,7 @@ XCOMM Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
 
 INCLUDES = -I$(TOP)
 DEPLIBS = $(DEPXLIB) $(DEPSMLIB)
-LOCAL_LIBRARIES = $(XLIB) $(SMLIB)
+LOCAL_LIBRARIES = $(XLIB) $(SMLIB) -lICE
 DEFINES = -DSHAPE
 
 HEADERS = lwm.h ewmh.h
-- 
1.7.0

