From 0fc576e69bc1dcbadacc1e0005c577be50916e3d Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Fri, 5 Mar 2010 11:27:09 +0200
Subject: [PATCH] lwm.c: no hard coded locale


Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 lwm.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/lwm.c b/lwm.c
index a0c0f8c..612131a 100644
--- a/lwm.c
+++ b/lwm.c
@@ -86,8 +86,6 @@ main(int argc, char *argv[]) {
 
 	mode = wm_initialising;
 
-	setlocale(LC_ALL,"");
-
 	/* Open a connection to the X server. */
 	dpy = XOpenDisplay(NULL);
 	if (dpy == 0)
-- 
1.7.0

