From 77713b317d9d6e00ba8f257f1db95f636fe0e329 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Fri, 5 Mar 2010 11:28:43 +0200
Subject: [PATCH] lwm.c: Use Debian terminal setting


Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 lwm.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/lwm.c b/lwm.c
index 61298b1..9648d1d 100644
--- a/lwm.c
+++ b/lwm.c
@@ -299,7 +299,7 @@ shell(ScreenInfo * screen, int button, int x, int y) {
 		execl(sh, sh, "-c", command, NULL);
 		fprintf(stderr, "%s: can't exec \"%s -c %s\"\n", argv0, sh,
 			command);
-		execlp("xterm", "xterm", NULL);
+		execlp("x-terminal-emulator", "x-terminal-emulator", NULL);
 		exit(EXIT_FAILURE);
 	case -1:	/* Error. */
 		fprintf(stderr, "%s: couldn't fork\n", argv0);
-- 
1.7.0

